lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -504,6 +576,20 @@ unsafeMkEd25519 :: XPub -> Ed25519.PublicKey
 unsafeMkEd25519 =
     throwCryptoError . Ed25519.publicKey . xpubPublicKey
 
+mkByronWitness
+    :: SL.TxBody TPraosStandardCrypto
+    -> ProtocolMagic
+    -> (XPrv, Passphrase "encryption")
+    -> SL.BootstrapWitness TPraosStandardCrypto
+mkByronWitness body protocolMagic (prv, _pwd) =
+    let signingKey = Crypto.SigningKey prv
+        bytes = xprvToBytes prv
+        addrAttr = Byron.mkAttributes $ Byron.AddrAttributes
+            (Just $ Byron.HDAddressPayload bytes)
```

Comment:
I don't think we should send the user's private key over in a witness! 
The HDAddressPayload corresponds to the encrypted derivation path that is present only on Byron-Random addresses. I assume that what we needs here is: 

a) To pass down the corresponding address to `mkByronWitness`
b) Have a function in the Compatibility module to convert an `Address` into a `AddrAttributes`. We have all the information we need in the Address already.

Reply:
have a try to deliver proper solution in https://github.com/input-output-hk/cardano-wallet/pull/1849/commits/72c70d0f90a40ac481bdbebd48199d67f0745163